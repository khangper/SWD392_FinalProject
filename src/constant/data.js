export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
