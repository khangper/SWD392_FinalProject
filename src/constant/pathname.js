export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
