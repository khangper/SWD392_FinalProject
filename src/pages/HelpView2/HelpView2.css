.HelpView-vew121 {
	font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
	text-align: left;
	line-height: 26px;
	color: #333;
}

.HelpView-vew121:hover {
	text-decoration: underline !important;
	color: #333;
}

.mt-50{
    margin-top: 50px;
}

.mb-30{
    margin-bottom: 30px;
}