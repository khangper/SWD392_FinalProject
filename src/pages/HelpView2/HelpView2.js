import React from "react";
import "./HelpView2.css";
import search_icon from "..//../assets/search.png";
import Payment from "..//../assets/payment.png";
import tablerow from "..//../assets/tablerow.png";
import helpdestop from "..//../assets/helpdestop.png";
import rulerhelp from "..//../assets/ruler-help.png";
import trade from "..//../assets/trade.png";
import bookhelp from "..//../assets/book-help.png";
import Header from "../../components/Header-paidmember/Header";

function HelpView2() {
  return (
    <div className="HelpView2">
      <Header />

      <div className="HelpView1-grid-container">
        <div className="HelpView-title125">
          <div className="HelpView-titleleft">
            <div className="HelpView-ttl121">
              <nav aria-label="breadcrumb">
                <ol className="HelpView-breadcrumb">
                  <li className="HelpView-breadcrumb-item">
                    <a href="index.html">Home</a>
                  </li>
                  <li className="HelpView-breadcrumb-item">
                    <a href="help.html">/Selling & Promotion</a>
                  </li>
                  <li
                    className="HelpView-breadcrumb-item active"
                    aria-current="page"
                  >
                    /Pricing & Coupons
                  </li>
                </ol>
              </nav>
            </div>
          </div>
          <div className="explore-container">
            <section className="search-box">
              <img
                src={search_icon}
                alt="Search Icon"
                className="search-icon"
              />
              <input type="text" placeholder="Search Tutors" />
            </section>
          </div>
        </div>
        <div className="HelpView-title126">
          <h2>Payments</h2>
        </div>
      </div>
      {/* Body HelpView2 */}

      <div className="HelpView-grid-container">
        <div className="HelpView-grid-item">
          <div className="HelpView-fcrse_3 HelpView-frc123">
            <div className="HelpView-cater_ttle">
              <h4>Instructor Topics</h4>
            </div>
            <ul className="HelpView-allcate15">
              <li>
                <a href="#" className="HelpView-ct_item active">
                  <img
                    src={Payment}
                    alt="Search Icon"
                    className="HelpView-icon"
                  />
                  Payments
                </a>
              </li>
              <li>
                <a href="#" className="HelpView-ct_item ">
                  <img
                    src={tablerow}
                    alt="Search Icon"
                    className="HelpView-icon"
                  />
                  Selling & Promotion
                </a>
              </li>
              <li>
                <a href="#" className="HelpView-ct_item">
                  <img
                    src={helpdestop}
                    alt="Search Icon"
                    className="HelpView-icon"
                  />
                  Quality Standards
                </a>
              </li>
              <li>
                <a href="#" className="HelpView-ct_item">
                  <img
                    src={rulerhelp}
                    alt="Search Icon"
                    className="HelpView-icon"
                  />
                  Course Building
                </a>
              </li>
              <li>
                <a href="#" className="HelpView-ct_item">
                  <img
                    src={trade}
                    alt="Search Icon"
                    className="HelpView-icon"
                  />
                  Course Management
                </a>
              </li>
              <li>
                <a href="#" className="HelpView-ct_item">
                  <img
                    src={bookhelp}
                    alt="Search Icon"
                    className="HelpView-icon"
                  />
                  Trust & Safety
                </a>
              </li>
            </ul>
          </div>
        </div>

        <div className="HelpView-grid-item">
          <div className="HelpView-vew120 HelpView-frc123">
            <h4>Taxes</h4>
            <div className="HelpView-ntt125">
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  Common Instructor Questions About The W-8BEN
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  Taxes: What do Instructors Need to Submit to Cursus?
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  What Does Cursus do With my Tax Information (i.e.
                  Withholding)?
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  How do I Submit my Tax Forms to Cursus?
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  How Does Cursus Handle VAT / GST?
                </a>
              </div>
            </div>
          </div>
          <div className="HelpView-vew120 mt-50 mb-30">
            <h4>Payments - General</h4>
            <div className="HelpView-ntt125">
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  How do I Earn Revenue From Cursus for Business?
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  How to opt out of Cursus for Business
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  Payoneer FAQ
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  Missing Instructor Payments
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  Instructor Revenue Share
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  Instructor Revenue Report
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  Instructor Payment Overview
                </a>
              </div>
              <div>
                <a href="help_faq_view.html" className="HelpView-vew121">
                  How do I Opt Into the Affiliate Program?
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default HelpView2;
