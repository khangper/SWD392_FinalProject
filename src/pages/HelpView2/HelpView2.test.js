import React from "react";
import { render, screen } from "@testing-library/react";
import HelpView2 from "./HelpView2";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";

describe("HelpView2", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the HelpView2 component", () => {
    const { container } = renderWithProviders(<HelpView2 />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it("displays the breadcrumb correctly", () => {
    renderWithProviders(<HelpView2 />);
    const breadcrumb = screen.getByText("/Pricing & Coupons");
    expect(breadcrumb).toBeInTheDocument();
  });

  it("displays the search input box correctly", () => {
    renderWithProviders(<HelpView2 />);
    const searchInput = screen.getByPlaceholderText("Search Tutors");
    expect(searchInput).toBeInTheDocument();
  });
});
