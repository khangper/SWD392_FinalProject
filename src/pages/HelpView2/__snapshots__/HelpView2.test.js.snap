// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpView2 renders the HelpView2 component 1`] = `
<div>
  <div
    class="HelpView2"
  >
    <header
      class="header1"
    >
      <a
        href="/home"
      >
        <button
          class="header-link1"
        >
          Back To Cursus
        </button>
      </a>
      <a
        href="/home"
      >
        <img
          alt=""
          class="header-group-icon"
          src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo.svg"
        />
      </a>
      <img
        alt=""
        class="header-avatar"
        src="profile-img.jpg"
      />
    </header>
    <div
      class="HelpView1-grid-container"
    >
      <div
        class="HelpView-title125"
      >
        <div
          class="HelpView-titleleft"
        >
          <div
            class="HelpView-ttl121"
          >
            <nav
              aria-label="breadcrumb"
            >
              <ol
                class="HelpView-breadcrumb"
              >
                <li
                  class="HelpView-breadcrumb-item"
                >
                  <a
                    href="index.html"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="HelpView-breadcrumb-item"
                >
                  <a
                    href="help.html"
                  >
                    /Selling & Promotion
                  </a>
                </li>
                <li
                  aria-current="page"
                  class="HelpView-breadcrumb-item active"
                >
                  /Pricing & Coupons
                </li>
              </ol>
            </nav>
          </div>
        </div>
        <div
          class="explore-container"
        >
          <section
            class="search-box"
          >
            <img
              alt="Search Icon"
              class="search-icon"
              src="search.png"
            />
            <input
              placeholder="Search Tutors"
              type="text"
            />
          </section>
        </div>
      </div>
      <div
        class="HelpView-title126"
      >
        <h2>
          Payments
        </h2>
      </div>
    </div>
    <div
      class="HelpView-grid-container"
    >
      <div
        class="HelpView-grid-item"
      >
        <div
          class="HelpView-fcrse_3 HelpView-frc123"
        >
          <div
            class="HelpView-cater_ttle"
          >
            <h4>
              Instructor Topics
            </h4>
          </div>
          <ul
            class="HelpView-allcate15"
          >
            <li>
              <a
                class="HelpView-ct_item active"
                href="#"
              >
                <img
                  alt="Search Icon"
                  class="HelpView-icon"
                  src="payment.png"
                />
                Payments
              </a>
            </li>
            <li>
              <a
                class="HelpView-ct_item "
                href="#"
              >
                <img
                  alt="Search Icon"
                  class="HelpView-icon"
                  src="tablerow.png"
                />
                Selling & Promotion
              </a>
            </li>
            <li>
              <a
                class="HelpView-ct_item"
                href="#"
              >
                <img
                  alt="Search Icon"
                  class="HelpView-icon"
                  src="helpdestop.png"
                />
                Quality Standards
              </a>
            </li>
            <li>
              <a
                class="HelpView-ct_item"
                href="#"
              >
                <img
                  alt="Search Icon"
                  class="HelpView-icon"
                  src="ruler-help.png"
                />
                Course Building
              </a>
            </li>
            <li>
              <a
                class="HelpView-ct_item"
                href="#"
              >
                <img
                  alt="Search Icon"
                  class="HelpView-icon"
                  src="trade.png"
                />
                Course Management
              </a>
            </li>
            <li>
              <a
                class="HelpView-ct_item"
                href="#"
              >
                <img
                  alt="Search Icon"
                  class="HelpView-icon"
                  src="book-help.png"
                />
                Trust & Safety
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="HelpView-grid-item"
      >
        <div
          class="HelpView-vew120 HelpView-frc123"
        >
          <h4>
            Taxes
          </h4>
          <div
            class="HelpView-ntt125"
          >
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                Common Instructor Questions About The W-8BEN
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                Taxes: What do Instructors Need to Submit to Cursus?
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                What Does Cursus do With my Tax Information (i.e. Withholding)?
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                How do I Submit my Tax Forms to Cursus?
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                How Does Cursus Handle VAT / GST?
              </a>
            </div>
          </div>
        </div>
        <div
          class="HelpView-vew120 mt-50 mb-30"
        >
          <h4>
            Payments - General
          </h4>
          <div
            class="HelpView-ntt125"
          >
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                How do I Earn Revenue From Cursus for Business?
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                How to opt out of Cursus for Business
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                Payoneer FAQ
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                Missing Instructor Payments
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                Instructor Revenue Share
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                Instructor Revenue Report
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                Instructor Payment Overview
              </a>
            </div>
            <div>
              <a
                class="HelpView-vew121"
                href="help_faq_view.html"
              >
                How do I Opt Into the Affiliate Program?
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
